packages/craco/lib/features/test/jest-babel-transform.js

```diff
@@ -1,38 +1,26 @@
-const babelJest = require("babel-jest");
-
 const { loadCracoConfig } = require("../../config");
-const { isArray } = require("../../utils");
-
-const craBabelTransformer = {
-    presets: ["babel-preset-react-app"],
-    babelrc: false,
-    configFile: false
-};
-
-const context = {
-    env: process.env.NODE_ENV
-};
-
-const cracoConfig = loadCracoConfig(context);
-
-const { addPresets, addPlugins } = cracoConfig.jest.babel;
-
-if (cracoConfig.babel) {
-    if (addPresets) {
-        const { presets } = cracoConfig.babel;
-
-        if (isArray(presets)) {
-            craBabelTransformer.presets = craBabelTransformer.presets.concat(presets);
+const { createJestBabelTransform } = require("./create-jest-babel-transform");
+
+let jestBabelTransform;
+
+// cracoConfig is only available inside the transform, but the transform needs to include whatever options cracoConfig
+// specifies. So, the first time this transform is run, it generates a new transform -- using cracoConfig -- and
+// uses that to process files.
+module.exports = {
+    ...createJestBabelTransform(),
+    process(src, filename, config, transformOptions) {
+        if (!jestBabelTransform) {
+            let cracoConfig = config.globals && config.globals._cracoConfig;
+            if (!cracoConfig) {
+                const context = {
```

Comment:
Could you extract the context out of this function?

Reply:
Done: context is no longer needed after removing the craco config load in the transform script (from your next comment)